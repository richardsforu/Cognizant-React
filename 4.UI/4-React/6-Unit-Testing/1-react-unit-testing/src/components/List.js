

import React from 'react';
import PropTypes from 'prop-types';
import ListItem from './ListItem'

function List(props) {
    const { items } = props;
    if (items.length === 0) {
        return (<span className="alert alert-info">No items in List</span>);
    }
    return (
        <div>
            <ul className="list-group">
                {items.map(item => <ListItem key={item} item={item} />)}
            </ul>
        </div>
    )
}

List.propTypes = {
    items: PropTypes.array
}
List.defaultProps = {
    items: []
}

export default List;