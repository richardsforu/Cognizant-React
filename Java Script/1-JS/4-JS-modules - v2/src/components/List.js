

class List {

}

class ListTheme {

}

class ListConfig {

}
export { ListTheme}


