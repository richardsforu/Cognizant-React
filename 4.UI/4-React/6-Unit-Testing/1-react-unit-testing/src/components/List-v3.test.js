
import React from 'react';
import List from './List';

import renderer from 'react-test-renderer'


describe('List components - tests', () => {

    test('render list', () => {
        let items = ["one", "two", "three"];
        const component = renderer.create(<List items={items} />)
        expect(component).toMatchSnapshot();
    });

});

// to update specic snapshot

// jest u -t="TestName"