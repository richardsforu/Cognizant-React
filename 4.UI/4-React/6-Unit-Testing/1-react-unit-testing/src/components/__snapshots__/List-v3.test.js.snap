// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List components - tests render list 1`] = `
<div>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
    >
      one
    </li>
    <li
      className="list-group-item"
    >
      two
    </li>
    <li
      className="list-group-item"
    >
      three
    </li>
  </ul>
</div>
`;
